great_expectations/data_context/store/checkpoint_store.py

```diff
@@ -227,18 +231,20 @@ def _persist_checkpoint(
         key: GXCloudIdentifier | ConfigurationIdentifier,
         checkpoint: Checkpoint,
         persistence_fn: Callable,
-    ) -> Checkpoint:
+    ) -> Checkpoint | CheckpointConfig:
         checkpoint_ref = persistence_fn(key=key, value=checkpoint.get_config())
         if isinstance(checkpoint_ref, GXCloudResourceRef):
-            # update parts of config that may have been updated by cloud (ids, default actions, etc.)
-            cloud_id = checkpoint_ref.id
-            checkpoint.config.ge_cloud_id = cloud_id
-            checkpoint.config.validations = checkpoint_ref.response["data"][
-                "attributes"
-            ]["checkpoint_config"].get("validations")
-            checkpoint.config.action_list = checkpoint_ref.response["data"][
-                "attributes"
-            ]["checkpoint_config"].get("action_list")
+            # return CheckpointConfig from cloud POST response to account for any defaults/new ids added in cloud
+            checkpoint_config = checkpoint_ref.response["data"]["attributes"][
+                "checkpoint_config"
+            ]
+            checkpoint_config["ge_cloud_id"] = checkpoint_config.pop("id")
```

Comment:
Are we guaranteed to have "id" in `checkpoint_config`? Will this `pop()` possible cause a runtime issue?

Reply:
Yeah, this should always be returned by cloud.